### PR TITLE
feature/move url to constant and change safe call

### PR DESCRIPTION
**Pull request by vov4ik-v**

### CODE REVIEW

File: app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/enums/TypeDevice.kt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.spm.vasylyshyn.counterdisplayapp.enums
+
+enum class TypeDevice {
+    GAS, WATER,LIGHT
```
Comment (BogdanDidukh2003) [inline app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/enums/TypeDevice.kt]:
```suggestion
    GAS, WATER, LIGHT
```

Comment (BogdanDidukh2003) [reply app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/enums/TypeDevice.kt]: Or

File: app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/enity/DisplayCount.kt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package com.spm.vasylyshyn.counterdisplayapp
+package com.spm.vasylyshyn.counterdisplayapp.enity
 
 import java.time.LocalDateTime
```
Comment (BogdanDidukh2003) [inline app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/enity/DisplayCount.kt]:
```suggestion
data class DisplayCount(
    val id: Int = 0,
    val displayCount: Int = 0,
    val createdDate: LocalDateTime = LocalDateTime.now(),
)
Minor: can we use a primary constractor with data class instead?
```

File: app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/enums/TypeDevice.kt (hunk 1 above)
Comment (BogdanDidukh2003) [inline app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/enums/TypeDevice.kt]:
```suggestion
    GAS,
    WATER,
    LIGHT,
```

File: app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/enity/Device.kt [hunk 3]
```diff
@@ -1,5 +1,6 @@
-package com.spm.vasylyshyn.counterdisplayapp
+package com.spm.vasylyshyn.counterdisplayapp.enity
 
+import com.spm.vasylyshyn.counterdisplayapp.enums.TypeDevice
 import java.time.LocalDateTime
 
 class Device {
```
Comment (BogdanDidukh2003) [inline app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/enity/Device.kt]: The same here: data class, with primary constructor, with default parameters

File: app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/fragment/StatisticsFragment.kt [hunk 4]
```diff
@@ -90,7 +95,7 @@ class StatisticsFragment : Fragment() {
             )
             device.listDisplayCounts = counts
 //            device.listDisplayCounts = map // edit
-                    arrayDevice.add(device)
+//                    arrayDevice.add(device) edit
```
Comment (BogdanDidukh2003) [inline app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/fragment/StatisticsFragment.kt]: Don't commit commented out code: delete it instead

File: app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/adapter/CardScrollAdapter.kt [hunk 5]
```diff
@@ -35,7 +39,11 @@ class CardScrollAdapter internal constructor(context: Activity?, devices: List<D
...
             if (currentDevice.typeDevice === TypeDevice.GAS) {
                 typeDeviceImage.setImageResource(R.drawable.gas_type)
-            } else {
+            }
+            if(currentDevice.typeDevice == TypeDevice.LIGHT){
+                typeDeviceImage.setImageResource(R.drawable.light_type)
+            }
+            if (currentDevice.typeDevice == TypeDevice.WATER) {
                 typeDeviceImage.setImageResource(R.drawable.water_type)
             }
```
Comment (BogdanDidukh2003) [inline app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/adapter/CardScrollAdapter.kt]:
```
            val typeDeviceImageResource = when (currentDevice.typeDevice) {
                TypeDevice.GAS -> R.drawable.gas_type
                TypeDevice.WATER -> R.drawable.water_type
                TypeDevice.LIGHT -> R.drawable.light_type
                null -> null
            }
            typeDeviceImageResource?.also { typeDeviceImage.setImageResource(it) }
```

Comment (BogdanDidukh2003) [reply app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/adapter/CardScrollAdapter.kt]: The last line is equialent to 
```
            if (typeDeviceImageResource != null) {
                typeDeviceImage.setImageResource(typeDeviceImageResource)
            }
```

File: app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/activity/MainActivity.kt [hunk 6]
```diff
@@ -52,23 +56,22 @@ class MainActivity : AppCompatActivity() {
...
                     .enqueue(object : Callback<User> {
                         override fun onResponse(
                             call: Call<User>,
                             response: Response<User>
                         ) {
                             if (response.isSuccessful) {
-                                username = response.body()!!.username.toString()
-                                phoneNumber = response.body()!!.phoneNumber.toString()
-                                userId = response.body()!!.id.toInt()
+                                username = response.body()?.username.toString()
```
Comment (BogdanDidukh2003) [inline app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/activity/MainActivity.kt:72]: image
I see that there are still ```!!``` operators in the project, can you double check them too please? They might be sources of potential app crashes

File: app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/activity/LoginActivity.kt [hunk 7]
```diff
@@ -34,9 +36,8 @@ class LoginActivity : AppCompatActivity() {
                 ) { dialog, whichw -> dialog.dismiss() }
                 alertDialog.show()
             } else {
-                val ip = IP.ip
                 val retrofit = Retrofit.Builder()
-                    .baseUrl("http://${ip}:8080/api/")
+                    .baseUrl(URL_PATH)
                     .addConverterFactory(GsonConverterFactory.create())
                     .build()
 //                val loginRequest:LoginRequest = LoginRequest("vasiliwin@gmail.com", "vova")
```
Comment (BogdanDidukh2003) [inline app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/activity/LoginActivity.kt]: Remove commented out code

**Review by BogdanDidukh2003 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/activity/LoginActivity.kt (hunk 7 above)
Comment (BogdanDidukh2003) [reply app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/activity/LoginActivity.kt]: Also, consider renaming URL_PATH to API_URL_PATH, or something even more specific. There might be plenty of urls used in the app, as it grows

Comment (vov4ik-v) [reply app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/activity/LoginActivity.kt]: Remove all commented out code and rename URL_PATH

File: app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/activity/MainActivity.kt (hunk 6 above)
Comment (vov4ik-v) [reply app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/activity/MainActivity.kt:72]: Replace all !! to ?

**Pull request merged**